proc/breakpoints.go

```
@@ -109,15 +114,25 @@ func (dbp *Process) writeSoftwareBreakpoint(thread *Thread, addr uint64) error {
 	return err
 }
 
-func (bp *Breakpoint) checkCondition(thread *Thread) bool {
-	if bp.Cond < 0 {
-		return true
+func (bp *Breakpoint) checkCondition(thread *Thread) (met bool, err error) {
+	if bp.Cond == nil {
+		return true, nil
 	}
-	g, err := thread.GetG()
+	scope, err := thread.Scope()
 	if err != nil {
-		return false
+		return true, nil
```

Why not return the error here?